app/main.py

```diff
@@ -1,2 +1,73 @@
+from typing import Any, Hashable
+from dataclasses import dataclass, field
+
+
+@dataclass
+class Node:
+    key: Hashable
+    hash_val: int
+    value: Any
+
+
+@dataclass
 class Dictionary:
-    pass
+    initial_capacity = 8
+    load_factor = 2 / 3
+    size = 0
+    hash_table: list[list[Node]] = field(init=False)
+
+    def __post_init__(self) -> None:
+        self.hash_table = [None] * self.initial_capacity
+
+    def line_stop(self) -> int:
+        return int(self.initial_capacity * self.load_factor)
+
+    def get_index(self, key: Hashable) -> int:
+        return hash(key) % self.initial_capacity
+
+    @staticmethod
+    def get_hash(key: Hashable) -> int:
+        return hash(key)
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size >= self.line_stop():
+            self.resize()
+
+        hash_val = self.get_hash(key)
+        index = self.get_index(key)
+
+        if self.hash_table[index] is None:
+            self.hash_table[index] = Node(key, hash_val, value)
+            self.size += 1
+            return
+        while self.hash_table[index]:
+            if self.hash_table[index].key == key:
+                self.hash_table[index].value = value
+                return
+            index = (index + 1) % self.initial_capacity
+            if self.hash_table[index] is None:
+                self.hash_table[index] = Node(key, hash_val, value)
+                self.size += 1
+                return
```

Comment:
```suggestion
        while True:
            if self.hash_table[index] is None:
                self.hash_table[index] = Node(key, hash_val, value)
                self.size += 1
                break

            if self.hash_table[index].key == key:
                self.hash_table[index].value = value
                break

            index = (index + 1) % self.initial_capacity
```

I think it's more clear